### ISSUE TITLE
WebhookStore/OffloadTargetStore destructor's leak-on-timeout escape hatch is defeated by StoreWorkerPool's own unconditional join

### ISSUE THREAD

**Issue by Doomgoose** [Tr3kkR/Yuzu#3297]
## Context

`WebhookStore`'s and `OffloadTargetStore`'s destructors (added/hardened in PR #3294) give up
after a 24h `pool_.quiesce()` timeout, log `"...leaking the SQLite handle rather than risking a
use-after-free..."`, and return — intending to leak and continue rather than block forever.

That doesn't actually happen. `pool_` is the last-declared member in both store headers, so its
own destructor runs immediately after the store's destructor body returns:

```cpp
~StoreWorkerPool() {
    quiesce_and_join_unconditional();  // untimed w.join() on every still-joinable worker
}
```

`quiesce_and_join_unconditional()` does an **untimed** `join()` on any worker still executing a
task — including the exact worker whose hung delivery just caused the 24h `quiesce()` above to
time out. So the destructor doesn't leak-and-continue as its log message promises; it logs, then
blocks forever at that `join()`.

**Not reachable via the production `ServerImpl::stop()` path today** — `stop()` quiesces with a
60s bound and `std::_Exit(1)`s on timeout before these destructors ever run naturally, and no
delivery work is in flight during `ServerImpl`'s own construction either (confirmed during
#3294's review). The only paths that would hit this are test code (which the shipped
`test_store_worker_pool.cpp` tests avoid via a `ReleaseGuard` that unblocks the task before the
pool destructs) or a hypothetical future standalone use of either store outside `ServerImpl` —
both explicitly named as real targets by the destructor's own comment.

## Acceptance criteria

- [ ] Either give `quiesce_and_join_unconditional()` a bounded, timed join with its own documented
      escape hatch (accepting the same `std::terminate`-on-uncooperative-thread risk `w.detach()`
      would introduce, as a stated tradeoff), or correct the destructor's log message/comment so
      it doesn't promise a graceful leak-and-continue that the pool's own teardown doesn't
      deliver.
